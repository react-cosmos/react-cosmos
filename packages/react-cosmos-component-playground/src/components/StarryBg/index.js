import React, { Component } from 'react';
import styles from './index.less';

class StarryBg extends Component {
  render() {
    const { children } = this.props;
    return (
      <div className={styles.root}>
        <div className={styles.stars}></div>
        {children &&
          <div className={styles.content}>
            {children}
          </div>}
      </div>
    );
  }
}

export default StarryBg;
