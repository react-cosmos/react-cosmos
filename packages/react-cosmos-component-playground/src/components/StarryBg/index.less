@import '../../utils/common.less';

@stars-texture: url('../../static/stars.png');

.root {
  .stretch();

  overflow: hidden;
  background-color: darken(@left-nav-bgcolor, 5%);
}

.stars {
  position: absolute;
  bottom: 0;
  right: 0;
  width: 300%;
  height: 300%;
  min-width: 4000px;
  min-height: 2750px;
  background-image: @stars-texture;
  background-position: center center;
  animation: stars-ani 200s linear infinite;
}

.content {
  .stretch();

  display: flex;
  justify-content: center;
  align-items: center;
}

@keyframes stars-ani {
  0% {
    transform: translateY(0) translateX(0);
  }
  100% {
    transform: translateY(1372px) translateX(2000px);
  }
}
