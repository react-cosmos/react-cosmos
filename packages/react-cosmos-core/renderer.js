module.exports = require('./dist/renderer/index.js');
