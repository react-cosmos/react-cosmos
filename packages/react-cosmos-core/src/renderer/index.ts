export * from './FixtureLoader/getDecoratedFixtureElement/index.js';
export * from './getFixtureList.js';
export * from './getSortedDecoratorsForFixturePath.js';
export * from './reactTypes.js';
export * from './registerPlaygroundShortcuts.js';
export * from './rendererConfig.js';
export * from './types.js';
