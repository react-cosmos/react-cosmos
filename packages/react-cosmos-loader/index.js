// @flow

exports.mount = require('./src/mount').mount;
exports.createContext = require('./src/create-context').createContext;
exports.Loader = require('./src/components/Loader').Loader;
