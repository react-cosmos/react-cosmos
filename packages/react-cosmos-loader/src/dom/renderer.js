// @flow

import { render, unmountComponentAtNode } from 'react-dom';

import type { LoaderWebOpts } from 'react-cosmos-flow/loader';
import type { Renderer } from 'react-cosmos-flow/context';

export function createDomRenderer(opts?: LoaderWebOpts = {}): Renderer {
  const { containerQuerySelector } = opts;
  const container = getDomContainer(containerQuerySelector);

  return function renderer(element) {
    render(element, container);

    return {
      unmount() {
        unmountComponentAtNode(container);
      },
      toJSON: () => null
    };
  };
}

function getDomContainer(querySelector?: string): HTMLElement {
  if (!querySelector) {
    return createDomContainer();
  }

  const existingContainer = document.querySelector(querySelector);
  if (!existingContainer) {
    console.warn(
      `[Cosmos] Could not find ${querySelector} element. Created fresh DOM container.`
    );
    return createDomContainer();
  }

  return existingContainer;
}

function createDomContainer() {
  const existingNode = document.getElementById('root');
  if (existingNode) {
    return existingNode;
  }

  const newNode = document.createElement('div');
  newNode.setAttribute('id', 'root');
  if (document.body) {
    document.body.appendChild(newNode);
  }

  return newNode;
}
