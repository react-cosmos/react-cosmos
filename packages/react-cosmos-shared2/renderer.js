module.exports = require('./dist/renderer');
