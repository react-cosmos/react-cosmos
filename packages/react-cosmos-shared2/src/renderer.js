// @flow

export const RENDERER_MESSAGE_EVENT_NAME = 'cosmos-cmd';
