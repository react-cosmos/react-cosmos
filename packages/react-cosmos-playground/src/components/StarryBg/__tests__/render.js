import React from 'react';
import { Loader } from 'react-cosmos-loader';
import renderer from 'react-test-renderer';
import StarryBg from '../';
import blankFixture from '../__fixtures__/blank';
import contentFixture from '../__fixtures__/content';

describe('StarryBg blank', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(
      <Loader component={StarryBg} fixture={blankFixture} />
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe('StarryBg content', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(
      <Loader component={StarryBg} fixture={contentFixture} />
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
