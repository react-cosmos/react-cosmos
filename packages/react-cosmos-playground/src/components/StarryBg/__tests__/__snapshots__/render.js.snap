// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarryBg blank renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="stars"
  />
</div>
`;

exports[`StarryBg content renders correctly 1`] = `
<div
  className="root"
>
  <div
    className="stars"
  />
  <div
    className="content"
  >
    floating over the starry sky...
  </div>
</div>
`;
